### PR TITLE
W.I.P: PHP 8.4: Add ReflectionConstant stubs

### PR DESCRIPTION
**Pull request by Girgias**
@iluuu1994 if you could fill in the docs

### CODE REVIEW

**Comment by iluuu1994** [comment]
@Girgias May I ask you to review this one as well? I found some inconsistencies in the pages I copied from. Namely, `<refpurpose>` is quite variable in how lengthy it is. Can you clarify whether a concise or detailed description is preferred?

File: reference/reflection/reflectionconstant/getfilename.xml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  &reftitle.parameters;
+  &no.function.parameters;
+ </refsect1>
+
+ <refsect1 role="returnvalues">
+  &reftitle.returnvalues;
+  <simpara>
+   Returns the filename of the file in which the constant has been defined. If
+   the constant is defined in the PHP core or in a PHP extension, &false; is
+   returned.
```
Comment (Girgias) [inline reference/reflection/reflectionconstant/getfilename.xml:30]: Semantic line breaks :)
```suggestion
   Returns the filename of the file in which the constant has been defined.
   If the constant is defined in the PHP core or in a PHP extension,
   &false; is returned.
```

File: reference/reflection/reflectionconstant/isdeprecated.xml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+var_dump((new ReflectionConstant('E_STRICT'))->isDeprecated());
+?>
+]]>
+   </programlisting>
+   &example.outputs;
+   <screen>
+<![CDATA[
+bool(true)
+]]>
+   </screen>
```
Comment (Girgias) [inline reference/reflection/reflectionconstant/isdeprecated.xml:51]: Could do this to showcase the difference in output depending on the version.
```suggestion
   &example.outputs.84;
   <screen>
<![CDATA[
bool(true)
]]>
   </screen>
   &example.outputs.83;
   <screen>
<![CDATA[
bool(false)
]]>
   </screen>
```

File: reference/reflection/reflectionconstant/tostring.xml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ </refnamediv>
+
+ <refsect1 role="description">
+  &reftitle.description;
+  <methodsynopsis role="ReflectionConstant">
+   <modifier>public</modifier> <type>string</type><methodname>ReflectionConstant::__toString</methodname>
+   <void/>
+  </methodsynopsis>
+  <simpara>
+   Returns the string representation of the ReflectionConstant object.
```
Comment (Girgias) [inline reference/reflection/reflectionconstant/tostring.xml:15]:
```suggestion
   Returns the string representation of the <classname>ReflectionConstant</classname> object.
```

**Review by Girgias — COMMENTED** [review]
Minor comments, refpurposes should be short-ish, as they appear 
image
next to the link to visit the page

**Comment by iluuu1994** [comment]
Done.

File: reference/reflection/reflectionconstant/isdeprecated.xml (hunk 2 above)
Comment (Girgias) [reply reference/reflection/reflectionconstant/isdeprecated.xml]: Ah I forgot that

**Comment by Girgias** [comment]
Thank you!

**Pull request merged**